src/components/Application-Header/favorite-list/FavoriteList.tsx

```diff
@@ -8,30 +8,52 @@ import StarIcon from '@material-ui/icons/Star';
 import Popover from '@material-ui/core/Popover';
 import Divider from '@material-ui/core/Divider';
 
+import { Advertisement } from '../../../types/Advertisements';
+import { RootStore } from '../../../store/RootStore';
 import FavoriteItem from './FavoriteItem';
-import { JobsStore } from '../../../store/JobsStore';
 import styles from './FavoriteListStyles';
 
 interface FavoriteListProps extends WithStyles<typeof styles> {
-    jobsStore?: JobsStore;
+    rootStore?: RootStore;
 }
 
 const FavoriteList: React.FC<FavoriteListProps> = (props): JSX.Element => {
     const { classes } = props;
-    const jobsStore: JobsStore = props.jobsStore!;
+    const rootStore: RootStore = props.rootStore!;
+    const userFavoriteAdsIds: number[] = rootStore.userStore.getUser.favoriteAds!;
+    const allAd: Advertisement[] = rootStore.jobsStore.advertisements;
 
     const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
     const handleSavedJobsButtonClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
-        setAnchorEl(event.currentTarget);
-      };
+        if (userFavoriteAdsIds.length > 0) {
+            setAnchorEl(event.currentTarget);
+        }
+    };
     
-      const handleClose = (): void => {
+    const handleClose = (): void => {
         setAnchorEl(null);
-      };
+    };
+
+    const getIconColor = (): string => {
+        return userFavoriteAdsIds.length > 0 ? classes.starIconYellow : classes.starIconWhite;
+    }
+
+    const displayFavoriteItems = (): JSX.Element[] => {
+        return userFavoriteAdsIds.map(favorId => {
```

Comment:
Since the logic is:
* Array of favorites
* All adds

Instead of checking for every favoriteAdd if you can find it (So you can display it),

I would:

`allAd.filter(ad => userFavoriteAdsIds.includes(ad.id))`

What do you think? I believe it is shorter 😃

Reply:
👍

Reply:
Done